src/main/java/stashpullrequestbuilder/stashpullrequestbuilder/StashRepository.java

```diff
@@ -205,13 +205,26 @@ public boolean mergePullRequest(String pullRequestId, String version)
     }
 
     private Boolean isPullRequestMergable(StashPullRequestResponseValue pullRequest) {
-        if (trigger.isCheckMergeable() || trigger.isCheckNotConflicted()) {
+        if (trigger.isCheckMergeable() || trigger.isCheckNotConflicted() || trigger.isCheckProbeMergeStatus()) {
             StashPullRequestMergableResponse mergable = client.getPullRequestMergeStatus(pullRequest.getId());
             boolean res = true;
             if (trigger.isCheckMergeable())
                 res = res && mergable.getCanMerge();
             if (trigger.isCheckNotConflicted())
                 res = res && !mergable.getConflicted();
+
+            if (trigger.isCheckProbeMergeStatus()) {
+                /* Just probe the REST API, so Stash updates the refspecs
+                 *
+                 * Workaround for broken git references that appear due to PRs,
+                 * popping up in other jobs trying to use just the master branch.
+                 *
+                 * See https://issues.jenkins-ci.org/browse/JENKINS-35219 and
+                 * https://community.atlassian.com/t5/Bitbucket-questions/Change-pull-request-refs-after-Commit-instead-of-after-Approval/qaq-p/194702
+                 */
+                return true;
+            }
```

Comment:
This `if` block of code seems unnecessary and incorrect to me. It the user requests specific checks, they should be performed, and their results should not be ignored. If this block is removed, the code would work correctly. There would be a call to `getPullRequestMergeStatus` (the whole point of this PR), the relevant checks ("can be merged" and "not conflicted") would be performed if enabled, the PR would be considered mergeable if the enabled checks pass or no checks are enabled.

Reply:
Thanks, your point does sound valid, especially if those other checks were requested and did fail.

I'll convert the `if` to just a big comment about why that toggle is consulted in the outer big `if` but not inside the `res` evaluation.